client/dashboard/sites/site/error.tsx

```diff
@@ -0,0 +1,54 @@
+import { useQueryClient } from '@tanstack/react-query';
+import { Notice } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import UnknownError from '../../app/500';
+import { siteBySlugQuery } from '../../app/queries/site';
+import { sitesQuery } from '../../app/queries/sites';
+import { siteRoute } from '../../app/router';
+import { PageHeader } from '../../components/page-header';
+import PageLayout from '../../components/page-layout';
+import RouterLinkButton from '../../components/router-link-button';
+import { DashboardDataError } from '../../data/error';
+import { getSiteDisplayName } from '../../utils/site-name';
+import type { Site } from '../../data/types';
+
+export default function Error( { error }: { error: Error } ) {
+	switch ( error instanceof DashboardDataError && error.code ) {
+		case 'inaccessible_jetpack':
+			return <InaccessibleJetpackError error={ error } />;
+		default:
+			return <UnknownError error={ error } />;
+	}
+}
+
+function InaccessibleJetpackError( { error }: { error: Error } ) {
+	const { siteSlug } = siteRoute.useParams();
+	const queryClient = useQueryClient();
+
+	let site = queryClient.getQueryData< Site >( siteBySlugQuery( siteSlug ).queryKey );
+	if ( ! site ) {
+		const allSites = queryClient.getQueryData< Site[] >( sitesQuery().queryKey );
+		site = allSites?.find( ( s ) => s.slug === siteSlug );
+	}
```

Comment:
Being careful to only try showing the display name if it happens to already be available. We don't want to cause more errors in the error handler!

Reply:
I feel this is a bit unnecessary. If the site is down, I don't think it's that helpful to show the site name; the site URL makes more sense to highlight.

Reply:
Yes perhaps you're right. I'm imagining that there might be other errors we could catch that are site related, that wouldn't be as catastrophic as this one. And in that case it'd be useful to salvage whatever `Site` data we have.

But we should cross that bridge when we come to it. In this case where the JP site has just disappeared it is a bit more catastrophic.